force-app/main/default/classes/GooglePlaces.cls

```diff
@@ -1,42 +1,137 @@
 public class GooglePlaces {
 
- 	public static void searchPlace(Decimal longitude, Decimal latitude, Integer radius, String type, String key){
+ 	public static void searchPlace (Id account, Decimal longitude, Decimal latitude, Integer radius, String type, String key){
         
+        // Validation to check if type is church. In the future it would be nice to add functionality to take in different types like School.
         if (type != 'church') {
             System.debug('Type needs to be church.');
             return;
         }
         
-        String googleNerbyPlacesUrl = 'https://maps.googleapis.com/maps/api/place/textsearch/json'+ '?location=' + longitude + ',' + latitude + '&radius=' + radius + '&type=' + type + '&key=' + key;
-        System.debug(googleNerbyPlacesUrl);
-        
+        // Using the textsearch call instead of nearbyplaces since the former provides formatted addresses. 
+        String googleNearbyPlacesUrl = 'https://maps.googleapis.com/maps/api/place/textsearch/json'+ '?location=' + longitude + ',' + latitude + '&radius=' + radius + '&type=' + type + '&key=' + key;
         Http http = new Http();
         HttpRequest request = new HttpRequest();
-        request.setEndpoint(googleNerbyPlacesUrl);
+        request.setEndpoint(googleNearbyPlacesUrl);
         request.setMethod('GET');
         HttpResponse response = http.send(request);
-        // If the request is successful, parse the JSON response.
+        
+        // If the request is successful, parse the JSON response. Status code is always 200 for Google Maps API
         if (response.getStatusCode() == 200) {
+            
             // Deserialize the JSON string into collections of primitive data types.
-            Map<String, Object> googleresults = (Map<String, Object>) JSON.deserializeUntyped(response.getBody());
+            Map<String, Object> googleResults = (Map<String, Object>) JSON.deserializeUntyped(response.getBody());
+            Map<Integer, Map<String,Object>> bulkData = new Map<Integer, Map<String,Object>>();
             // Cast the values in the 'animals' key as a list
-            List<Object> results = (List<Object>) googleresults.get('results');
+            List<Object> results = (List<Object>) googleResults.get('results');
             System.debug('Received the following results:');
-            for (Object result: results) {
-                System.debug(result);
-                System.debug('formatting');
-                Map<String,Object> formatted = (Map<string,Object>) result;
-                System.debug(formatted.get('formatted_address'));
-                System.debug(formatted.get('name'));
-
+            
+            for (Integer i = 0; i < results.size(); i ++) {
+                
+            	// Setting the limit of records created to 5 for now. In the future, can add functionality to dynamically create a requested number of records.
+                if (i == 5) {
+                    break;
```

Comment:
Does this break stop all iterations or just the current iteration?

If just current iteration, consider setting the value of i to results.size() before break
i.e.
if (i ==5) {
  i = results.size();
 break;
}